src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Picker/ContentPicker/ContentPickerDialog.jsx

```diff
@@ -39,59 +95,67 @@ const ContentPickerDialog = ({
-            */
-
-                        const tableConfig = {
-                            typeFilter: pickerConfig.selectableTypesTable,
-                            recursionTypesFilter: ['nt:base'],
-                            showOnlyNodesWithTemplates: pickerConfig.showOnlyNodesWithTemplates
-                        };
-
-                        return (
-                            <ContentTable
-                                tableConfig={tableConfig}
-                                setSelectedItem={setSelectedItem}
-                                selectedPath={selectedPath}
-                                initialSelection={initialSelection}
-                                editorContext={editorContext}
-                            />
-                        );
-                    }}
-                </PickerDialog>
-            )}/>
+            <FastField shouldUpdate={() => true}
+                       render={({form: {setFieldValue, setFieldTouched}}) => (
+                           <PickerDialog
+                               displayTree={pickerConfig.displayTree}
+                               idInput={id}
+                               site={site}
+                               siteNodes={siteNodes}
+                               lang={editorContext.lang}
+                               initialSelectedItem={initialSelectedItem}
+                               nodeTreeConfigs={nodeTreeConfigs.map(nodeTreeConfig => ({
+                                   ...nodeTreeConfig,
+                                   rootPath: nodeTreeConfig.treeConfig.rootPath(site),
+                                   rootLabel: siteNode.displayName
+                               }))}
+                               modalCancelLabel={t('content-editor:label.contentEditor.edit.fields.modalCancel').toUpperCase()}
+                               modalDoneLabel={t('content-editor:label.contentEditor.edit.fields.modalDone').toUpperCase()}
+                               onSelectSite={siteNode => onSelectSite(siteNode)}
+                               onCloseDialog={() => setIsOpen(false)}
+                               onItemSelection={content => {
+                                   setFieldValue(
+                                       id,
+                                       content[0] ? content[0].id : null,
+                                       true
+                                   );
+                                   setIsOpen(false);
+                                   setFieldTouched(field.name, field.multiple ? [true] : true);
+                               }}
+                           >
+                               {(setSelectedItem, selectedPath, initialSelection) => {
+                                   // Build table config from picker config
+                                   /*
+                      Todo: make the picker work as CMM, use the recursionTypesFilter to browse all contents within a page
```

Comment:
A TODO should be linked to a task in jira :)

Reply:
(even it's not your code :/ )

Reply:
You're the owner of this code hahaha, can you please add the jira task for that?